Remove deprecated `:example_group` example metadata sub-hash (#2851)
rspec/rspec-core: Changelog.md

* Raise on usage of metadata on suite-level scopes. (Phil Pirozhkov, #2849)
* Raise an error when `fail_fast` is configured with
  an unsupported value. (Phil Pirozhkov, #2849)
* Remove deprecated exposition of example group's metadata

>   in example's metadata `:example_group` subhash. (Phil Pirozhkov, #2851)

Reviewer: This needs rewording but I havent had enough coffee yet to process how.

> * Remove deprecated exposition of example group's metadata
>   in example's metadata `:example_group` subhash. (Phil Pirozhkov, #2851)

Reviewer: ```suggestion
* Remove deprecated access to an example group's metadata through the example.
  (Phil Pirozhkov, #2851)
```

Author: Thanks! Much better this way.
